File: src/main/java/se/gewalli/AppendBatch.java
package se.gewalli;

import io.atlassian.fugue.Either;
import se.gewalli.commands.Command;

import java.util.Collection;
import java.util.concurrent.CompletableFuture;

public interface AppendBatch {
    CompletableFuture<Either<FailureReason, Integer>> batch(Collection<Command> commands);

    CompletableFuture<Either<FailureReason, Collection<Command>>> readAll();
}

File: src/main/java/se/gewalli/AppendBatch.kt
package se.gewalli

import se.gewalli.commands.Command
import se.gewalli.kyminon.Result
import java.util.concurrent.CompletableFuture

interface AppendBatch {
    fun batch(commands: Collection<Command>): CompletableFuture<Result<Int, FailureReason>>
    fun readAll(): CompletableFuture<Result<Collection<Command>, FailureReason>>
}